tel2puml/pipelines/data_creation.py:19: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:52: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:106: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:107: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:119: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:148: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:161: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tel2puml/utils.py:77: error: Missing type parameters for generic type "tuple"  [type-arg]
tel2puml/utils.py:78: error: Name "plt.Figure" is not defined  [name-defined]
tel2puml/sequence_otel.py:14: error: Untyped decorator makes function "get_slice_indexes" untyped  [misc]
tel2puml/sequence_otel.py:31: error: Untyped decorator makes function "has_time_overlap" untyped  [misc]
tel2puml/detect_loops.py:207: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[Loop]")  [assignment]
tel2puml/detect_loops.py:464: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[str]")  [assignment]
tel2puml/check_puml_equiv.py:54: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/check_puml_equiv.py:130: error: Need type annotation for "collected_attributes" (hint: "collected_attributes: Dict[<type>, <type>] = ...")  [var-annotated]
tel2puml/check_puml_equiv.py:229: error: Argument 1 to "create_networkx_graph_from_parsed_puml" has incompatible type "list[str]"; expected "list[Any | tuple[Literal['START', 'END', 'PATH'], Literal['XOR', 'AND', 'OR', 'LOOP']]]"  [arg-type]
tel2puml/check_puml_equiv.py:229: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tel2puml/check_puml_equiv.py:229: note: Consider using "Sequence" instead, which is covariant
tel2puml/check_puml_equiv.py:281: error: Returning Any from function declared to return "bool"  [no-any-return]
tests/test_sequence_otel.py:157: error: Argument 1 to "add_span" of "Trace" has incompatible type "*tuple[object, ...]"; expected "str"  [arg-type]
tests/test_sequence_otel.py:157: error: Argument 1 to "add_span" of "Trace" has incompatible type "*tuple[object, ...]"; expected "float"  [arg-type]
tests/test_sequence_otel.py:157: error: Argument 1 to "add_span" of "Trace" has incompatible type "*tuple[object, ...]"; expected "str | None"  [arg-type]
tests/test_check_puml_equiv.py:33: error: Need type annotation for "collected_attributes" (hint: "collected_attributes: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_check_puml_equiv.py:104: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tests/test_check_puml_equiv.py:217: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:218: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:221: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:222: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:257: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:258: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:261: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:262: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:285: error: Incompatible types in assignment (expression has type "bool", target has type "str")  [assignment]
tests/test_check_puml_equiv.py:286: error: Incompatible types in assignment (expression has type "bool", target has type "str")  [assignment]
tel2puml/logic_detection.py:407: error: Argument 1 to "remove" of "set" has incompatible type "set[Any]"; expected "frozenset[str]"  [arg-type]
tel2puml/logic_detection.py:410: error: Argument 1 to "get_weighted_cover" has incompatible type "set[frozenset[str]]"; expected "set[set[Any] | frozenset[Any]]"  [arg-type]
tel2puml/logic_detection.py:471: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/events.py:41: error: Signature of "__hash__" incompatible with supertype "dict"  [override]
tel2puml/events.py:41: note:      Superclass:
tel2puml/events.py:41: note:          None
tel2puml/events.py:41: note:      Subclass:
tel2puml/events.py:41: note:          def __hash__(self) -> int
tel2puml/puml_graph/graph.py:55: error: "type[PUMLOperatorNodes]" has no attribute "PATH_LOOP"  [attr-defined]
tel2puml/puml_graph/graph.py:78: error: Argument "extra_info" to "__init__" of "NXNode" has incompatible type "dict[str, bool] | None"; expected "dict[str, str] | None"  [arg-type]
tel2puml/puml_graph/graph.py:296: error: Class cannot subclass "DiGraph" (has type "Any")  [misc]
tel2puml/puml_graph/graph.py:390: error: Incompatible return value type (got "tuple[PUMLOperatorNode, ...]", expected "tuple[PUMLOperatorNode, PUMLOperatorNode]")  [return-value]
tel2puml/puml_graph/graph.py:533: error: Invalid index type "PUMLNode" for "dict[PUMLEventNode | PUMLOperatorNode, list[PUMLEventNode | PUMLOperatorNode]]"; expected type "PUMLEventNode | PUMLOperatorNode"  [index]
tel2puml/puml_graph/graph.py:539: error: Call to untyped function (unknown) in typed context  [no-untyped-call]
tel2puml/puml_graph/graph.py:551: error: Incompatible return value type (got "list[PUMLNode]", expected "list[PUMLEventNode | PUMLOperatorNode]")  [return-value]
tel2puml/puml_graph/graph.py:649: error: Returning Any from function declared to return "PUMLGraph"  [no-any-return]
tests/puml_graph/test_graph.py:47: error: Non-overlapping equality check (left operand type: "dict[str, str]", right operand type: "dict[str, bool]")  [comparison-overlap]
tests/puml_graph/test_graph.py:311: error: Non-overlapping equality check (left operand type: "dict[str, str]", right operand type: "dict[str, bool]")  [comparison-overlap]
tests/puml_graph/test_graph.py:320: error: Non-overlapping equality check (left operand type: "dict[str, str]", right operand type: "dict[str, bool]")  [comparison-overlap]
tests/puml_graph/test_graph.py:533: error: Need type annotation for "node_set" (hint: "node_set: Set[<type>] = ...")  [var-annotated]
tests/puml_graph/test_graph.py:579: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:580: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:581: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/puml_graph/test_graph.py:606: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:608: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:609: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:611: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/puml_graph/test_graph.py:613: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/puml_graph/test_graph.py:639: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:641: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:642: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:644: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/puml_graph/test_graph.py:646: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tel2puml/puml_graph/graph_loop_extract.py:56: error: Need type annotation for "start_loop_nodes" (hint: "start_loop_nodes: Set[<type>] = ...")  [var-annotated]
tel2puml/puml_graph/graph_loop_extract.py:57: error: Need type annotation for "end_loop_nodes" (hint: "end_loop_nodes: Set[<type>] = ...")  [var-annotated]
tel2puml/pipelines/data_ingestion.py:13: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_ingestion.py:25: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "dict_values[str, list[dict[Any, Any]]]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tel2puml/pipelines/data_ingestion.py:120: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_ingestion.py:134: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_ingestion.py:140: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node.py:90: error: Unsupported left operand type for + ("None")  [operator]
tel2puml/node_map_to_puml/node.py:90: note: Left operand is of type "str | None"
tel2puml/node_map_to_puml/node.py:282: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node.py:284: error: Incompatible default for argument "uid" (default has type "None", argument has type "str")  [assignment]
tel2puml/node_map_to_puml/node.py:284: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tel2puml/node_map_to_puml/node.py:284: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tel2puml/node_map_to_puml/node.py:285: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node.py:285: error: Incompatible default for argument "incoming" (default has type "None", argument has type "list[Any]")  [assignment]
tel2puml/node_map_to_puml/node.py:285: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tel2puml/node_map_to_puml/node.py:285: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tel2puml/node_map_to_puml/node.py:286: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node.py:286: error: Incompatible default for argument "outgoing" (default has type "None", argument has type "list[Any]")  [assignment]
tel2puml/node_map_to_puml/node.py:286: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tel2puml/node_map_to_puml/node.py:286: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tel2puml/node_map_to_puml/node.py:287: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node.py:287: error: Incompatible default for argument "incoming_logic" (default has type "None", argument has type "list[Any]")  [assignment]
tel2puml/node_map_to_puml/node.py:287: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tel2puml/node_map_to_puml/node.py:287: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tel2puml/node_map_to_puml/node.py:288: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node.py:288: error: Incompatible default for argument "outgoing_logic" (default has type "None", argument has type "list[Any]")  [assignment]
tel2puml/node_map_to_puml/node.py:288: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tel2puml/node_map_to_puml/node.py:288: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tel2puml/node_map_to_puml/node.py:360: error: Incompatible return value type (got "tuple[PUMLEvent, ...]", expected "tuple[PUMLEvent]")  [return-value]
tel2puml/node_map_to_puml/node.py:361: error: Incompatible return value type (got "tuple[Never, ...]", expected "tuple[PUMLEvent]")  [return-value]
tel2puml/node_map_to_puml/node.py:495: error: Need type annotation for "event_node_ref" (hint: "event_node_ref: Dict[<type>, <type>] = ...")  [var-annotated]
tel2puml/node_map_to_puml/node.py:588: error: Incompatible types in assignment (expression has type "None", variable has type "int")  [assignment]
tel2puml/node_map_to_puml/node.py:614: error: No overload variant of "__setitem__" of "list" matches argument types "int", "PUMLNode"  [call-overload]
tel2puml/node_map_to_puml/node.py:614: note: Possible overload variants:
tel2puml/node_map_to_puml/node.py:614: note:     def __setitem__(self, SupportsIndex, PUMLOperatorNode, /) -> None
tel2puml/node_map_to_puml/node.py:614: note:     def __setitem__(self, slice, Iterable[PUMLOperatorNode], /) -> None
tel2puml/node_map_to_puml/node.py:641: error: Incompatible types in assignment (expression has type "list[PUMLOperatorNode | PUMLNode]", variable has type "list[PUMLOperatorNode]")  [assignment]
tel2puml/node_map_to_puml/node.py:649: error: Incompatible return value type (got "tuple[PUMLNode, Node | None]", expected "tuple[PUMLNode, Node]")  [return-value]
tel2puml/node_map_to_puml/node.py:706: error: Argument 1 to "has_event_set_as_subset" of "Event" has incompatible type "list[str | None]"; expected "list[str]"  [arg-type]
tel2puml/node_map_to_puml/node.py:816: error: Argument 1 to "create_event_node" of "PUMLGraph" has incompatible type "str | None"; expected "str"  [arg-type]
tel2puml/node_map_to_puml/node.py:848: error: Incompatible types in assignment (expression has type "PUMLKillNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/node_map_to_puml/node.py:851: error: Incompatible types in assignment (expression has type "PUMLOperatorNode | PUMLEventNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/node_map_to_puml/node.py:866: error: Incompatible types in assignment (expression has type "PUMLNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/node_map_to_puml/node.py:880: error: Argument 2 to "update_puml_graph_with_event_node" has incompatible type "Node | None"; expected "Node"  [arg-type]
tel2puml/node_map_to_puml/node.py:891: error: Incompatible types in assignment (expression has type "PUMLNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/node_map_to_puml/node.py:902: error: Incompatible types in assignment (expression has type "PUMLOperatorNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/node_map_to_puml/node.py:937: error: Argument 1 to "create_operator_node_pair" of "PUMLGraph" has incompatible type "PUMLOperator | None"; expected "PUMLOperator"  [arg-type]
tel2puml/node_map_to_puml/node.py:947: error: Incompatible return value type (got "tuple[PUMLOperatorNode | PUMLEventNode, Node]", expected "tuple[PUMLOperatorNode, Node]")  [return-value]
tel2puml/node_map_to_puml/node.py:1010: error: Argument 2 to "create_logic_merge" of "LogicBlockHolder" has incompatible type "Node | None"; expected "Node"  [arg-type]
tel2puml/node_map_to_puml/node.py:1025: error: Argument 3 to "handle_rotate_path" has incompatible type "PUMLNode"; expected "PUMLOperatorNode"  [arg-type]
tel2puml/node_map_to_puml/node.py:1052: error: Incompatible types in assignment (expression has type "PUMLNode", variable has type "PUMLOperatorNode")  [assignment]
tel2puml/node_map_to_puml/node.py:1059: error: Incompatible types in assignment (expression has type "PUMLOperatorNode | PUMLEventNode", variable has type "PUMLOperatorNode")  [assignment]
tel2puml/node_map_to_puml/node.py:1119: error: Incompatible return value type (got "tuple[PUMLNode, Node]", expected "tuple[PUMLOperatorNode | PUMLEventNode, Node]")  [return-value]
tel2puml/node_map_to_puml/node.py:1159: error: Incompatible types in assignment (expression has type "PUMLEventNode", variable has type "PUMLOperatorNode")  [assignment]
tests/test_detect_loops.py:507: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[Loop]")  [assignment]
tests/test_detect_loops.py:529: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[Loop]")  [assignment]
tests/test_detect_loops.py:572: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:602: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:625: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:649: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:676: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:727: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:762: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:801: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/tel2puml/test_logic_detection.py:64: error: Argument 2 to "create_augmented_data_from_reduced_event_set" has incompatible type "EventSet"; expected "frozenset[str]"  [arg-type]
tests/tel2puml/test_logic_detection.py:224: error: Function is missing a type annotation  [no-untyped-def]
tests/tel2puml/test_logic_detection.py:243: error: Function is missing a type annotation  [no-untyped-def]
tests/tel2puml/test_logic_detection.py:259: error: Call to untyped function "_check_or" in typed context  [no-untyped-call]
tests/tel2puml/test_logic_detection.py:266: error: Call to untyped function "_check_or_and" in typed context  [no-untyped-call]
tests/tel2puml/test_logic_detection.py:460: error: Function is missing a return type annotation  [no-untyped-def]
tests/tel2puml/test_logic_detection.py:552: error: Function is missing a type annotation  [no-untyped-def]
tests/tel2puml/test_logic_detection.py:561: error: Call to untyped function "_check_node_for_repeat_and" in typed context  [no-untyped-call]
tests/tel2puml/test_logic_detection.py:581: error: Call to untyped function "_check_node_for_repeat_and" in typed context  [no-untyped-call]
tests/tel2puml/test_logic_detection.py:740: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/test_logic_detection.py:779: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/test_logic_detection.py:829: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/test_logic_detection.py:877: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/test_logic_detection.py:927: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/test_logic_detection.py:981: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/pipelines/test_data_ingestion.py:19: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/pipelines/test_data_ingestion.py:72: error: Argument 1 to "get_graph_solutions_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:58: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:64: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:75: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:81: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:303: error: Argument 1 to "remove" of "list" has incompatible type "tuple[Hashable, Hashable]"; expected "tuple[tuple[str, str, int], tuple[str, str, int]]"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:321: error: Argument 1 to "check_correct_start_and_end_nodes" of "TestGetLoopStartAndEnd" has incompatible type "list[tuple[PUMLNode, PUMLNode]]"; expected "list[tuple[PUMLEventNode, PUMLEventNode]]"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:336: error: Argument 1 to "check_correct_start_and_end_nodes" of "TestGetLoopStartAndEnd" has incompatible type "list[tuple[PUMLNode, PUMLNode]]"; expected "list[tuple[PUMLEventNode, PUMLEventNode]]"  [arg-type]
tests/node_map_to_puml/test_node.py:34: error: Module "tel2puml.puml_graph.graph" does not explicitly export attribute "PUMLOperatorNodes"  [attr-defined]
tests/node_map_to_puml/test_node.py:96: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:120: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:152: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:184: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:217: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:265: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:303: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:340: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:447: error: Argument 3 to "load_logic_tree_into_nodes" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:475: error: Argument 3 to "load_all_logic_trees_into_nodes" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:518: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/node_map_to_puml/test_node.py:551: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/node_map_to_puml/test_node.py:570: error: Need type annotation for "expected_incoming_event_type_logic"  [var-annotated]
tests/node_map_to_puml/test_node.py:586: error: Unsupported right operand type for in ("object")  [operator]
tests/node_map_to_puml/test_node.py:595: error: Value of type "object" is not indexable  [index]
tests/node_map_to_puml/test_node.py:654: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/node_map_to_puml/test_node.py:793: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/node_map_to_puml/test_node.py:816: error: Argument 1 to "rotate_path" of "LogicBlockHolder" has incompatible type "Node | None"; expected "Node"  [arg-type]
tel2puml/puml_graph/graph_loop_insert.py:42: error: Argument 1 to "insert_loops" has incompatible type "PUMLGraph | None"; expected "PUMLGraph"  [arg-type]
tests/puml_graph/test_graph_loop_insert.py:47: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph_loop_insert.py:57: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph_loop_insert.py:73: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph_loop_insert.py:79: error: Item "None" of "PUMLGraph | None" has no attribute "subgraph_nodes"  [union-attr]
tests/node_map_to_puml/test_node_update.py:106: error: Argument 1 to "update_logic_nodes_with_lonely_merges_from_node_to_node_kill_map" has incompatible type "dict[str, list[str]]"; expected "dict[str, Iterable[str]]"  [arg-type]
tests/node_map_to_puml/test_node_update.py:106: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tests/node_map_to_puml/test_node_update.py:106: note: Consider using "Mapping" instead, which is covariant in the value type
tel2puml/pv_to_puml.py:89: error: Argument 1 to "update_logic_nodes_with_lonely_merges_from_node_to_node_kill_map" has incompatible type "dict[str, set[str]]"; expected "dict[str, Iterable[str]]"  [arg-type]
tel2puml/pv_to_puml.py:89: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tel2puml/pv_to_puml.py:89: note: Consider using "Mapping" instead, which is covariant in the value type
tel2puml/pv_to_puml.py:136: error: Returning Any from function declared to return "list[PVEvent]"  [no-any-return]
tests/test_pv_to_puml.py:16: error: Argument 1 to "pv_to_puml_string" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_pv_to_puml.py:26: error: Argument 1 to "pv_to_puml_string" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_pv_to_puml.py:37: error: Argument 1 to "pv_to_puml_string" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_pv_to_puml.py:47: error: Argument 1 to "pv_to_puml_string" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tel2puml/utils_test.py:39: error: Argument 1 to "pv_to_puml_string" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
Found 168 errors in 25 files (checked 42 source files)
